ui/lib/kv/addon/components/page/secret/metadata/details.js

```diff
@@ -32,21 +35,21 @@ export default class KvSecretMetadataDetails extends Component {
     // metadata tab is available even if user only has access to kv/data path
     return this.args.metadata?.customMetadata || this.args.secret?.customMetadata;
   }
+
   @action
   async onDelete() {
     // The only delete option from this view is delete all versions
-    const { secret } = this.args;
```

Comment:
This previously always returned a model so we could call `destroyRecord`, however, now we only fetch the secret model as a backup for reading `custom_metadata` so we don't want to rely on it for calling the delete. I opted to make a custom method instead since we only need the backend and path